packages/eslint-plugin/src/rules/dependency-checks.ts

```diff
@@ -0,0 +1,325 @@
+    }
+    const fileName = normalizePath(context.getFilename());
+    // support only package.json and project.json
+    if (
+      !fileName.endsWith('/package.json') &&
+      !fileName.endsWith('/project.json')
+    ) {
+      return {};
+    }
+
+    const projectPath = normalizePath(
+      (global as any).projectPath || workspaceRoot
+    );
+    const sourceFilePath = getSourceFilePath(fileName, projectPath);
+    const { projectGraph, projectRootMappings, projectFileMap } =
+      readProjectGraph(RULE_NAME);
+
+    if (!projectGraph) {
+      return {};
+    }
+
+    const sourceProject = findProject(
+      projectGraph,
+      projectRootMappings,
+      sourceFilePath
+    );
+
+    // check if source project exists, it's library and has a build target
+    if (!sourceProject || sourceProject.type !== 'lib') {
+      return {};
+    }
+
+    // check if library has a build target
+    const buildTarget = buildTargets.find(
+      (t) => sourceProject.data.targets?.[t]
+    );
+    if (!buildTarget) {
+      return {};
+    }
+
+    // gather helper dependencies for @nx/js executors
+    const helperDependencies = getHelperDependenciesFromProjectGraph(
+      workspaceRoot,
+      sourceProject.name,
+      projectGraph
+    );
+
+    // find all dependencies for the project
+    const npmDeps = findProjectsNpmDependencies(
+      sourceProject,
+      projectGraph,
+      buildTarget,
+      {
+        helperDependencies: helperDependencies.map((dep) => dep.target),
+      },
+      removePackageJsonFromFileMap(projectFileMap)
+    );
+    const projDependencies = {
+      ...npmDeps.dependencies,
+      ...npmDeps.peerDependencies,
+    };
+    const projDependencyNames = Object.keys(projDependencies);
+
+    const projPackageJsonPath = join(
+      workspaceRoot,
+      sourceProject.data.root,
+      'package.json'
+    );
+    let projPackageJsonDeps: Record<string, string> = {};
+    if (existsSync(projPackageJsonPath)) {
+      projPackageJsonDeps = getAllDependencies(projPackageJsonPath);
+    }
+
+    const rootPackageJsonDeps = getAllDependencies(
+      join(workspaceRoot, 'package.json')
+    );
+
+    function validateMissingDependencies(node: AST.JSONProperty) {
+      if (checkMissingDependencies) {
+        const missingDeps = projDependencyNames.filter(
+          (d) => !projPackageJsonDeps[d]
+        );
+
+        missingDeps.forEach((d) => {
+          if (!ignoredDependencies.includes(d)) {
+            context.report({
+              node: node as any,
+              messageId: 'missingDependency',
+              data: { packageName: d },
+              fix(fixer) {
+                const deps = (node.value as AST.JSONObjectExpression)
+                  .properties;
+                if (deps.length) {
+                  return fixer.insertTextAfter(
+                    deps[deps.length - 1] as any,
+                    `,\n    "${d}": "${
+                      rootPackageJsonDeps[d] || projDependencies[d]
```

Comment:
What happens if the package is not defined in the root `package.json`? Can we skip the fix and let the user know to add it manually?

This could happen if the user imports a new package but hasn't updated either `package.json` files yet.

Reply:
If it's not installed yet, it would not show up. It would be treated the same way as e.g. `fs` or `child_process`.